fastlane/lib/fastlane/setup/setup.rb

```diff
@@ -1,60 +1,272 @@
+      if ios_projects.count > 0
+        current_directory = ios_projects.find_all do |current_project_path|
+          current_project_path.split(File::Separator).count == 1
+        end
+        chosen_project = nil
+        had_multiple_projects_to_choose_from = false
+
+        if current_directory.count == 1
+          chosen_project = current_directory.first
+        elsif current_directory.count > 1
+          if current_directory.count == 2
+            # This is a common case (e.g. with CocoaPods), where the project has an xcodeproj and an xcworkspace file
+            extensions = [File.extname(current_directory[0]), File.extname(current_directory[1])]
+            if extensions.sort == [".xcodeproj", ".xcworkspace"].sort
+              # Yep, that's this kind of setup
+              chosen_project = current_directory.find { |d| d.end_with?(".xcworkspace") }
+            end
+          end
+          chosen_project ||= UI.select("Multiple iOS projects found in current directory", current_directory)
+          had_multiple_projects_to_choose_from = true
+        else
+          UI.error("Looks like there is no iOS project in the current directory, but inside a sub-directory instead")
+          UI.error("Please use `cd` into the subfolder of the location of your Xcode project")
+          UI.user_error!("Please `cd` into the subfolder of your Xcode project and run `fastlane init` again")
+        end
+
+        if chosen_project == "Pods.xcodeproj"
+          unless UI.confirm("Found '#{chosen_project}', which usually isn't a valid Xcode project. Make sure to switch to the directory containing your iOS Xcode project. Do you still want to continue?")
+            UI.user_error!("Make sure to `cd` into the right directory and then use `fastlane init` again")
+          end
+        end
+        UI.message("Detected iOS/Mac project in current directory: '#{chosen_project}'")
+
+        SetupIos.new(
+          is_swift_fastfile: is_swift_fastfile,
+          user: user,
+          project_path: chosen_project,
+          had_multiple_projects_to_choose_from: had_multiple_projects_to_choose_from
+        ).setup_ios
+      elsif android_projects.count > 0
         UI.message("Detected Android project in current directory...")
-        platform = :android
-      elsif is_react_native?
-        UI.important("Detected react-native app. To set up fastlane, please run")
-        UI.command("fastlane init")
-        UI.important("in the sub-folder for each platform (\"ios\" or \"android\")")
-        UI.user_error!("Please navigate to the platform subfolder and run `fastlane init` again")
+        SetupAndroid.new.setup_android
+      else
+        UI.error("No iOS or Android projects found in current directory '#{Dir.pwd}'")
+        UI.error("Make sure to `cd` into a directory containing your iOS or Android app")
+        if UI.confirm("Do you still want to setup a manual fastlane config in the current directory?")
+          SetupIos.new(
```

Comment:
you could ask whether people want to setup ios or android by default. Right?

Reply:
Ah nice catch, will update. Just waiting for the language checks by @taquitos 👍